src/message_log.rs

```diff
@@ -384,25 +384,22 @@ mod tests {
         log.garbage_collect(5);
 
         for old in 1..3 {
-            for msg_type in &[
+            for msg_type in vec![
```

Comment:
I'm not familiar with the advantages of one or the other; if we use a slice here, the `msg_type` will just have to 
be deref'd. Can you or @peterschwarz explain what the impact would be one way or the other?

Reply:
This particular usage is in a test, so it's impact is low, but in many cases there's an allocation for a vec on the heap, versus a stack allocation, so there are different performance/memory implications for one over the other.

For this test, you should have to change the list, but you can (I think) do `*msg_type` when using the message type later.  This should work because it is `Copy`.

Reply:
Updated to use a slice